platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -0,0 +1,43 @@
+#import "NSExpression+MGLAdditions.h"
+
+@implementation NSExpression (MGLAdditions)
+
+- (std::vector<mbgl::Value>)mgl_filterValues
+{
+    if ([self.constantValue isKindOfClass:NSArray.class]) {
+        NSArray *values = self.constantValue;
+        std::vector<mbgl::Value>convertedValues;
+        for (id value in values) {
+            convertedValues.push_back([self mgl_convertedValueWithValue:value]);
+        }
+        return convertedValues;
+    }
+    [NSException raise:@"Values not handled" format:@""];
+    return { };
+}
+
+- (mbgl::Value)mgl_filterValue
+{
+    return [self mgl_convertedValueWithValue:self.constantValue];
+}
+
+- (mbgl::Value)mgl_convertedValueWithValue:(id)value
+{
+    if ([value isKindOfClass:NSString.class]) {
+        return { std::string([(NSString *)value UTF8String]) };
+    } else if ([value isKindOfClass:NSNumber.class]) {
+        NSNumber *number = (NSNumber *)value;
+        if((strcmp([number objCType], @encode(int))) == 0) {
+            return { number.intValue };
+        } else if ((strcmp([number objCType], @encode(double))) == 0) {
+            return { number.doubleValue };
+        } else if ((strcmp([number objCType], @encode(bool))) == 0) {
```

Comment:
It triggered the exception since it was not caught in the previous conditions. Fixed here https://github.com/mapbox/mapbox-gl-native/pull/6049/commits/eefe5cb050789078b7fae0f05f13fbc929731240